source/extensions/filters/network/dubbo_proxy/active_message.cc

```diff
@@ -0,0 +1,407 @@
+                                                  SerializationType serialization_type) {
+  request_id_ = message_id;
+  filter_action_ = [type, message_id, serialization_type](
+                       DubboFilters::DecoderFilter* filter) -> Network::FilterStatus {
+    return filter->messageBegin(type, message_id, serialization_type);
+  };
+
+  return this->applyDecoderFilters(nullptr);
+}
+
+Network::FilterStatus ActiveMessage::messageEnd(MessageMetadataSharedPtr metadata) {
+  ASSERT(metadata->message_type() == MessageType::Request ||
+         metadata->message_type() == MessageType::Oneway);
+
+  // Currently only hessian serialization is implemented.
+  ASSERT(metadata->serialization_type() == SerializationType::Hessian);
+
+  ENVOY_LOG(debug, "dubbo request: start processing downstream request messages, id is {}",
+            metadata->request_id());
+
+  metadata_ = metadata;
+  filter_action_ = [metadata](DubboFilters::DecoderFilter* filter) -> Network::FilterStatus {
+    return filter->messageEnd(metadata);
+  };
+
+  auto status = applyDecoderFilters(nullptr);
+  if (status == Network::FilterStatus::StopIteration) {
+    pending_message_end_ = true;
+    return status;
+  }
+
+  finalizeRequest();
+
+  ENVOY_LOG(debug, "dubbo request: complete processing of downstream request messages, id is {}",
+            request_id_);
+
+  return status;
+}
+
+Network::FilterStatus ActiveMessage::transferHeaderTo(Buffer::Instance& header_buf, size_t size) {
+  filter_action_ = [&header_buf,
+                    size](DubboFilters::DecoderFilter* filter) -> Network::FilterStatus {
+    return filter->transferHeaderTo(header_buf, size);
+  };
+
+  return applyDecoderFilters(nullptr);
+}
+
+Network::FilterStatus ActiveMessage::transferBodyTo(Buffer::Instance& body_buf, size_t size) {
+  filter_action_ = [&body_buf, size](DubboFilters::DecoderFilter* filter) -> Network::FilterStatus {
+    return filter->transferBodyTo(body_buf, size);
+  };
+
+  return applyDecoderFilters(nullptr);
+}
+
+void ActiveMessage::finalizeRequest() {
+  pending_message_end_ = false;
+  parent_.stats().request_.inc();
+  bool is_one_way = false;
+  switch (metadata_->message_type()) {
+  case MessageType::Request:
+    parent_.stats().request_twoway_.inc();
+    break;
+  case MessageType::Oneway:
+    parent_.stats().request_oneway_.inc();
+    is_one_way = true;
+    break;
+  default:
+    break;
+  }
+
+  if (local_response_sent_ || is_one_way) {
+    parent_.deferredMessage(*this);
+  }
+}
+
+void ActiveMessage::createFilterChain() {
+  parent_.config().filterFactory().createFilterChain(*this);
+}
+
+DubboProxy::Router::RouteConstSharedPtr ActiveMessage::route() {
+  if (cached_route_) {
+    return cached_route_.value();
+  }
+
+  if (metadata_ != nullptr) {
+    DubboProxy::Router::RouteConstSharedPtr route =
+        parent_.config().routerConfig().route(*metadata_, stream_id_);
+    cached_route_ = std::move(route);
```

Comment:
route is shared_ptr, you can copy it directly.

Reply:
done.